### DISCUSSION TITLE
Ubiquiti Removed Local Account Creation

### DISCUSSION THREAD

**Discussion by juchong** [danielfernau/unifi-protect-video-downloader#210] [Q&A] [answered]
It looks like Ubiquiti removed the ability to create a local user account that also has ssh access in their latest release. I'm now only able to log in using root / ssh-password. I'm trying to access footage on a UNVR. I also have a cloud key gen 2 that I'm no longer using for video recording.

**Comment by danielfernau** [comment]
Hi there!

I'm not sure what your question is here. You don't need to run the tool on the UDM/UNVR/CloudKey directly. Since it's primarily distributed as a Docker Image you can use any computer on your network that has the Docker Engine installed. From there just use the credentials created via the web interface of your UniFi appliance as per documentation to download the footage within your desired date and time range.

Let me know if you need additional help with any of these steps. 

But also, thanks for the hint. I'll have a look if they have changed anything related to SSH accounts etc.

**Reply by juchong** [reply]
Hi @danielfernau! Apologies for the "light" issue report. I haven't been able to get the video downloader script to communicate with a newly-flashed UNVR. Only the root account seems to work to log in via SSH, but unfortunately, the script doesn't seem to be happy with logging in with that account. Have you been able to reproduce the issue?

**Reply by danielfernau** [reply]
Hi @juchong - no worries. The script neither uses nor needs any kind of SSH access. All you need is a machine with Docker installed on which you can run the tool as well as a "Local user" created via the UDM/UNVR web interface with "Viewer" permissions for Protect:

Navigate to the Web UI > Admins > Add User
- Role: `Viewer`
- Account Type: `Local Access Only`
- Local Username: `choose-any-username`
- Local Password: `add-a-secret-password`

Finally, click "Add" in the bottom right corner.

Then use username and password as per documentation with the tool to download your video files.

**Reply by juchong** [reply]
@danielfernau what does your setup look like? There doesn't seem to be an "Admins" tab anywhere that I can find. Here's the error I'm running into: 

![image](https://user-images.githubusercontent.com/342107/188251762-ca5b7467-1b58-447e-9dc0-31f95b0b6fbe.png)

Keep in mind that I'm using a UNVR + UCKP, which means that the credentials may not work the same as a "single device" setup like the UDMP.

**Reply by danielfernau** [reply]
@juchong Thanks for the reply and additional details. I still think that there's an issue with the way your accounts are configured on the UNVR. The UCKP (if not used for UniFi Protect) doesn't matter in this scenario. The UNVR, however, needs to have a local account set up by opening its web interface in a browser and adding it using the steps detailed here: https://help.ui.com/hc/en-us/articles/1500011491541-UniFi-Manage-users-and-user-roles#add-invite-users 
These instructions apply to all UniFi OS appliances, not just the UDMP. Hope this helps - if not, let me know where you're having difficulties and I'll try to assist further.

**Comment by juchong** [comment, marked as answer]
@danielfernau I figured it out! It looks like the API only seems to work over the RJ45 port on the UNVR. The script failed to run if I pointed it at the IP address of the SFP+ port. ¯\_(ツ)_/¯

**Reply by danielfernau** [reply]
@juchong Interesting, thanks for the update! As far as I know, the SFP+ is for 10Gbit and the RJ-45 for 1Gbit connection to your network, whichever fits your setup best. The SFP+ cannot be used in 1Gbit mode, at least to my knowledge. There's also a possibility that you have to use _either_ port, not both at the same time, but I'm not entirely sure on that one either as I can't test it right now. As long as there are no additional VLANs and (potentially mismatching) firewall rules in place it _should_ work via either one of these ports. Might also just be a bug in the OS or config of the UNVR (or the switch)...
But glad to hear that it works now! 👍
